hazelcast/src/main/java/com/hazelcast/spi/discovery/impl/DefaultDiscoveryService.java

```diff
@@ -155,11 +155,17 @@ private boolean validateCandidate(DiscoveryNode candidate) {
 
             List<DiscoveryStrategy> discoveryStrategies = new ArrayList<DiscoveryStrategy>();
             for (DiscoveryStrategyFactory factory : factories) {
+                //as a side-effect the buildDiscoveryStrategy() removes a config of the created strategy
                 DiscoveryStrategy discoveryStrategy = buildDiscoveryStrategy(factory, discoveryStrategyConfigs);
                 if (discoveryStrategy != null) {
                     discoveryStrategies.add(discoveryStrategy);
                 }
             }
+            if (!discoveryStrategyConfigs.isEmpty()) {
```

Comment:
Can't we just compare `discoveryStrategies.size() < discoveryStrategyConfigs.size()` instead of modifying the config object?

Reply:
Any progress on this one?

Reply:
thanks for a reminder. I'll push an update today.